packages/node_modules/@node-red/editor-client/src/js/ui/editor.js

```diff
@@ -848,99 +842,89 @@ RED.editor = (function() {
     function updateConfigNodeSelect(name,type,value,prefix,filter) {
         // if prefix is null, there is no config select to update
         if (prefix) {
-            var button = $("#"+prefix+"-edit-"+name);
```

Comment:
Why have you removed this? It is unrelated to the fix and removes functionality for singleton config nodes - a super rare edge case, but an edge case all the same.

Reply:
This part still exists in `prepareConfigNodeButton`.

If it's a singleton config node, `prepareConfigNodeButton` is used otherwise it's `prepareConfigNodeSelect` is used.

Reply:
Right - but this is `updateConfigNodeSelect` - used when the associated config node has been updated - not when doing the initial prepare. So I'm not clear why this has been removed.

Reply:
`prepareConfigNodeButton` doesn't use `updateConfigNodeSelect` so I don't really understand why we should do this check - I've never seen this case (singleton) so it's possible that it works differently than I think.

Reply:
Is removing this code necessary to fix the bug? If not, then don't remove it as it for a scenario you are not familiar with.

Reply:
No, i will revert this part for now - I will look into this scenario in more depth later.

Reply:
Sorry, you are right, `updateConfigNodeSelect` is used to update the button after the node config has been created/modified.
I will rewrite it properly in my future PR on the config node because as it stands it's not coherent to do it there.